classlib/src/main/java/org/teavm/classlib/java/util/TTreeMap.java

```diff
@@ -609,10 +618,14 @@ public int size() {
                         }
                     }
                     size -= path.length;
+                    if (fromPath != null && fromPath.length > 0 && path.length > 0 && owner.comparator.compare(
+                            fromPath[fromPath.length - 1].getKey(), path[path.length - 1].getKey()) == 0) {
```

Comment:
I suppose `fromPath[fromPath.length - 1] == path[path.length - 1]` is enough here